app/controllers/api/v1/mixins/service_offering_mixin.rb

```diff
@@ -4,13 +4,22 @@ module Mixins
       module ServiceOfferingMixin
         private
 
-        def service_offering_check(order)
-          service_offering_service = Catalog::ServiceOffering.new(order).process
+        def service_offering_check
+          service_offering_service = Catalog::ServiceOffering.new(@order).process
           if service_offering_service.archived
-            raise Catalog::ServiceOfferingArchived, "Service offering for order #{order.id} has been archived and can no longer be ordered"
+            @order.order_items.each do |order_item|
+              order_item.update!(:completed_at => DateTime.now, :state => "Failed")
+              order_item.update_message("error", archived_error_message)
+            end
+
+            @order.update!(:state => "Failed")
```

Comment:
WIth the `mark_failed` changes now merged from #630, lets get that code in here.

Reply:
Tagging @lindgrenj6 
Hmm, just had a thought about this, right now the `order_item` is calling `OrderStateTransition` after marking itself, but my code is going through (potentially) multiple order items. In this case, if we had multiple order items, we'd be calling the `OrderStateTransition` multiple times for no real reason. How should we handle that? Do we care?

Reply:
@eclarizio Isn't that a generic way of doing it. Today we don't have multiple order items but when we go to using ITSM we would have multiple order items and that's where this would help us. If one order item fails it has to mark the order as failed.

Reply:
Personally I think it's ok the way it is - the only problem being the fact that we'll get a progress message for each order_item, but at the same time that isn't a bad thing either because we would have confirmation that each order_item got marked as failed and the order would be marked failed.

Reply:
> @eclarizio Isn't that a generic way of doing it. Today we don't have multiple order items but when we go to using ITSM we would have multiple order items and that's where this would help us. If one order item fails it has to mark the order as failed.

Yeah that's what it's doing now, except it's just doing it once after all the order items have been marked as failed. Jacob's `mark_failed` method will mark the order as failed every time a single order item is marked as failed. As he pointed out, though, it's probably not a big deal.